"panic: UNKNOWN *vector.Named" with count() by named type

```
$ super -version &&
echo '
  type zenum = string
  {proto:"tcp"::zenum}
  {proto:"udp"::zenum}
  {proto:"tcp"::zenum}
' | super -i sup -c 'count() by proto | sort proto' -

Version: v0.3.0-221-g2c7cf3b83

panic: UNKNOWN *vector.Named

goroutine 83 [running]:
github.com/brimdata/super/runtime/vam/op/aggregate.(*countByString).update(0x3aa6651a8ec0?, {0x3aa6651a8ea0?, 0x0?, 0x3aa6651a8eb0?}, {0x3aa6651a8eb0?, 0x1?, 0x0?})
	/Users/phil/work/super/runtime/vam/op/aggregate/aggtable.go:174 +0x3ed
github.com/brimdata/super/runtime/vam/op/aggregate.(*Aggregate).consume(0x3aa6652c9d90, {0x3aa6651a8ea0?, 0x10bd99a5?, 0x200a66530cf90?}, {0x3aa6651a8eb0?, 0x4000011d0bbd0?, 0x5cd63168?})
	/Users/phil/work/super/runtime/vam/op/aggregate/aggregate.go:112 +0x309
github.com/brimdata/super/runtime/vam/op/aggregate.(*Aggregate).Pull.func1({0x3aa6651a8ea0, 0x2, 0x1059c765?})
	/Users/phil/work/super/runtime/vam/op/aggregate/aggregate.go:89 +0x75
github.com/brimdata/super/vector.Apply(0x6, 0x3aa665039e60, {0x3aa6651a8ea0, 0x2, 0x2})
	/Users/phil/work/super/vector/apply.go:43 +0xe7
github.com/brimdata/super/runtime/vam/op/aggregate.(*Aggregate).Pull(0x3aa6652c9d90, 0xfb?)
	/Users/phil/work/super/runtime/vam/op/aggregate/aggregate.go:88 +0x129
github.com/brimdata/super/runtime/vam/op.(*combineParent).run(0x3aa665302800)
	/Users/phil/work/super/runtime/vam/op/combine.go:109 +0x3c
created by github.com/brimdata/super/runtime/vam/op.(*Combine).Pull.func1 in goroutine 39
	/Users/phil/work/super/runtime/vam/op/combine.go:42 +0x35

```

## Details

Repro is with super commit 2c7cf3b. This issue surfaced in an Autoperf run of the `count_group_by_type_enum` query against the wrccdc data set.

The problem surfaced with the merge of #7072, as the problem did not repro at commit a8dec8d right before that merged.

```
$ super -version && echo '
  type zenum = string
  {proto:"tcp"::zenum}
  {proto:"udp"::zenum}
  {proto:"tcp"::zenum}
' | super -i sup -c 'count() by proto | sort proto' -

Version: v0.3.0-220-ga8dec8db0

type zenum=string
{proto:"tcp"::zenum,count:2}
{proto:"udp"::zenum,count:1}
```

--- a/runtime/vam/op/aggregate/aggtable.go
+++ b/runtime/vam/op/aggregate/aggtable.go
@@ -161,7 +161,7 @@ func (c *countByString) update(keys, vals []vector.Any) {
 		c.updatePartial(keys[0], vals[0])
 		return
 	}
-	switch val := keys[0].(type) {
+	switch val := vector.Under(keys[0]).(type) {
 	case *vector.String:
 		c.count(val)
 	case *vector.Dict:

--- a/runtime/ztests/op/aggregate/count-by-string.yaml
+++ b/runtime/ztests/op/aggregate/count-by-string.yaml
@@ -1,0 +1,26 @@
+spq: count() by x | sort x
+
+input: |
+  type s = string
+  {x:"a"::s}
+  {x:"b"::s}
+  {x:"c"::s}
+  {x:"d"::s}
+  {x:"e"::s}
+  {x:"a"}
+  {x:"b"}
+  {x:"c"}
+  {x:"d"}
+  {x:"a"}
+  {x:"b"}
+  {x:"c"}
+  {x:"a"}
+  {x:"b"}
+  {x:"a"}
+
+output: |
+  {x:"a",count:5}
+  {x:"b",count:4}
+  {x:"c",count:3}
+  {x:"d",count:2}
+  {x:"e",count:1}

